### PR TITLE
feat(*): list of kms implementation

### PR DESCRIPTION
**Pull request by vovh**
ref: MANAGER-13564

| Question         | Answer
| ---------------- | ---
| Branch?          |  feat/key-management-service
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | yes
| Tickets          | Fix #MANAGER-13564
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

## Description

Currently, there is no Key Management Service implemented on the OVHcloud Manager.

A customer can’t create any key, credential or secret.

When a customer want to create a key, a credential or a secret, there is necessarily a wallet, called KMS, to stock all information.

The objective will be to propose this new feature, directly in the menu.

### CODE REVIEW

File: packages/manager/apps/container/vite.config.mjs [hunk 1]
```diff
@@ -1,3 +1,3 @@
 import { defineConfig } from 'vite';
 import { getBaseConfig } from '@ovh-ux/manager-vite-config';
-export default defineConfig(getBaseConfig());
+export default defineConfig(getBaseConfig({isLABEU: true}));
```
Comment (chipp972) [inline packages/manager/apps/container/vite.config.mjs:3]: Are you sure you want to push this ?

File: packages/manager/apps/key-management-service/src/api/GET/apiv2/services.ts [hunk 2]
```diff
@@ -93,3 +96,37 @@ export const getListingIcebergV2 = async ({
...
+    const List = await fetchIcebergV2({
+      route: '/okms/resource',
+    }).then(({ data, status }) => ({ data, status }));
+    return List.data as OKMS[];
+  } catch (error) {
+    return null;
+  }
+};
+
+export const filterOKMS = (okms: OKMS[], sorting: ColumnSort): OKMS[] => {
```
Comment (chipp972) [inline packages/manager/apps/key-management-service/src/api/GET/apiv2/services.ts:119]: it's called "filter" but it does a sort

File: packages/manager/apps/key-management-service/src/pages/listing/index.tsx [hunk 3]
```diff
@@ -1,105 +1,137 @@
...
-    );
-  }
+  const { error, data: okms, isLoading } = useOKMS({
+    sorting,
+  });
 
-  if (data?.length === 0) return <Navigate to="onboarding" />;
+  useEffect(() => {
+    if (okms.length === 0 && !isLoading) {
+      navigate('/onboarding');
```
Comment (chipp972) [inline packages/manager/apps/key-management-service/src/pages/listing/index.tsx:76]: You should import your routes from a constant

File: packages/manager/apps/key-management-service/vite.config.mjs [hunk 4]
```diff
@@ -1,4 +1,4 @@
 import { defineConfig } from 'vite';
 import { getBaseConfig } from '@ovh-ux/manager-vite-config';
 
-export default defineConfig(getBaseConfig());
+export default defineConfig(getBaseConfig({ isLABEU: true }));
```
Comment (chipp972) [inline packages/manager/apps/key-management-service/vite.config.mjs:4]: you should not push this either

File: packages/manager/apps/key-management-service/src/pages/listing/index.tsx [hunk 5]
```diff
@@ -1,105 +1,137 @@
...
-        )}
+      <OsdsDivider></OsdsDivider>
+      <Notifications />
+      <div className={'flex mb-3 mt-6'}>
+        <OsdsButton
+          className="mr-1"
+          size={ODS_BUTTON_SIZE.sm}
+          variant={ODS_BUTTON_VARIANT.stroked}
+          color={ODS_THEME_COLOR_INTENT.primary}
+          href={'/create'}
```
Comment (chipp972) [inline packages/manager/apps/key-management-service/src/pages/listing/index.tsx:107]: Put this route in a constant

File: packages/manager/apps/key-management-service/src/pages/listing/index.tsx [hunk 6]
```diff
@@ -1,105 +1,137 @@
...
+  const { error, data: okms, isLoading } = useOKMS({
+    sorting,
+  });
 
-  if (data?.length === 0) return <Navigate to="onboarding" />;
+  useEffect(() => {
+    if (okms.length === 0 && !isLoading) {
+      navigate('/onboarding');
+    }
+  }, [okms.length]);
```
Comment (chipp972) [inline packages/manager/apps/key-management-service/src/pages/listing/index.tsx:78]:
```suggestion
  }, [okms.length, isLoading]);
```

File: packages/manager/apps/key-management-service/src/pages/listing/index.tsx (hunk 3 above)
Comment (vovh) [reply packages/manager/apps/key-management-service/src/pages/listing/index.tsx]: Fixed

File: packages/manager/apps/key-management-service/vite.config.mjs (hunk 4 above)
Comment (vovh) [reply packages/manager/apps/key-management-service/vite.config.mjs]: Fixed

File: packages/manager/apps/key-management-service/src/pages/listing/index.tsx (hunk 5 above)
Comment (vovh) [reply packages/manager/apps/key-management-service/src/pages/listing/index.tsx]: Fixed

File: packages/manager/apps/key-management-service/src/pages/listing/index.tsx (hunk 6 above)
Comment (vovh) [reply packages/manager/apps/key-management-service/src/pages/listing/index.tsx]: fixed

File: packages/manager/apps/key-management-service/src/pages/listing/index.tsx [hunk 7]
```diff
@@ -1,105 +1,137 @@
...
   return (
     <>
-      <h2>key-management-service</h2>
-      <div className="text-right pr-6 pb-6">
-        <OsdsText color={ODS_THEME_COLOR_INTENT.text}>
-          ({flattenData.length} {t('results')})
+      <OsdsBreadcrumb
+        items={[
+          {
+            href: '/key-management-system/',
```
[the remark this answers is not in the document]
Comment (vovh) [reply packages/manager/apps/key-management-service/src/pages/listing/index.tsx]: Fixed

File: packages/manager/apps/key-management-service/src/api/GET/apiv2/services.ts (hunk 2 above)
Comment (vovh) [reply packages/manager/apps/key-management-service/src/api/GET/apiv2/services.ts]: Fixed

**Review by aboungnaseng-ovhcloud — APPROVED** [review]

**Review by chipp972 — APPROVED** [review]

**Pull request merged**